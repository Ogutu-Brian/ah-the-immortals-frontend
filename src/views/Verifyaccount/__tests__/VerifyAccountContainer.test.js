import React from 'react';
import { Router } from 'react-router-dom';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import toJson from 'enzyme-to-json';
import { Button } from 'react-bootstrap';
import VerifyAccount from '../VerifyAccountContainer';

describe('Tests VerifyAccountContainer', () => {
  const store = configureStore([thunk])({
    verification: {
      error: {
        response: {
          message: 'fail'
        }
      },
      isActivating: false }});
  const props = {
    verifyAccountAction: jest.fn(),
    match: {
      params: {
        token: 'hbchujhb8g7g48y2uby82'
      }
    },
    error: {
      response: {
        message: 'fail'
      }
    },
    isLoading: false,
    history: {
      push: jest.fn()
    }
  };
  const history = createBrowserHistory();
  const wrapper = mount (
    <Router history={history}>
      <Provider store={store}>
        <VerifyAccount {...props} />
      </Provider>
    </Router>
  );
  it('tests mounting of container', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should call handleReload method' , () => {
    const button = wrapper.find('button[id="try"]');
    button.simulate('click');
    expect(button.exists()).toEqual(true);
  });
});
