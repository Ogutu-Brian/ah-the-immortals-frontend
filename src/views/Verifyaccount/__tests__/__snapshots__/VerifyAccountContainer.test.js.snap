// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests VerifyAccountContainer tests mounting of container 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(VerifyAccount)
      error={
        Object {
          "response": Object {
            "message": "fail",
          },
        }
      }
      history={
        Object {
          "push": [MockFunction],
        }
      }
      isLoading={false}
      match={
        Object {
          "params": Object {
            "token": "hbchujhb8g7g48y2uby82",
          },
        }
      }
      verifyAccountAction={[MockFunction]}
    >
      <VerifyAccount
        error={
          Object {
            "response": Object {
              "message": "fail",
            },
          }
        }
        history={
          Object {
            "push": [MockFunction],
          }
        }
        isLoading={false}
        match={
          Object {
            "params": Object {
              "token": "hbchujhb8g7g48y2uby82",
            },
          }
        }
        verifyAccountAction={[Function]}
      >
        <div>
          <div
            className="account"
          >
            <h1>
              Account Activation
            </h1>
            <div>
              <div
                className="mb-3"
              >
                <p
                  className="acc-error"
                >
                  Oops! Something went wrong.
                </p>
                <Button
                  active={false}
                  disabled={false}
                  id="try"
                  onClick={[Function]}
                  type="button"
                  variant="danger"
                >
                  <button
                    className="btn btn-danger"
                    disabled={false}
                    id="try"
                    onClick={[Function]}
                    type="button"
                  >
                    TRY AGAIN!
                  </button>
                </Button>
              </div>
            </div>
          </div>
        </div>
      </VerifyAccount>
    </Connect(VerifyAccount)>
  </Provider>
</Router>
`;
